src/services/AppSettings/index.ts

```diff
@@ -0,0 +1,42 @@
+import { BehaviorSubject, Subscription, tap } from 'rxjs'
+import { PersistenceService, persistenceService } from '../PersistenceService'
+import CONFIG from '../../config'
+
+export const NAMESPACE_APP_SETTINGS = 'appSettings'
+
+export const KEY_DEFAULT_LANGUAGE = `${CONFIG.CHAIN_TYPE}_${NAMESPACE_APP_SETTINGS}_defaultLanguage`
+export const KEY_DISPLAY_DEPRECATED_ADDRESSES = `${NAMESPACE_APP_SETTINGS}_displayDeprecatedAddresses`
+
+export type LanuageType = 'en' | 'zh'
+
+export class AppSettings {
+  defaultLanguage$ = new BehaviorSubject(this.persistenceService.get<LanuageType>(KEY_DEFAULT_LANGUAGE, 'en'))
+
+  displayDeprecatedAddresses$ = new BehaviorSubject(
+    this.persistenceService.get<boolean>(KEY_DISPLAY_DEPRECATED_ADDRESSES, false),
+  )
```

Comment:
I would suggest setting `show address in deprecated format` in URL so the page can be shared with specific address format info

Reply:
This should also require creating a new issue to handle, as it is a new feature rather than a refactor.

Reply:
https://github.com/Magickbase/ckb-explorer-public-issues/issues/467